cmds/webboot/webboot_test.go

```diff
@@ -133,31 +133,34 @@ func TestDirOption(t *testing.T) {
 	}
 }
 
-// This test should be run in the sudo mode.
-// The test need a usb stick that contains a /Image folder
-// which structure is
-// +-- Image
-// |  +-- tinycore1
-// |  |  +-- TinyCorePure64.iso
-// |  |  +-- TinyCorePure64 (copy).iso
-// |  +-- tinycore2
-// |     +-- TinyCorePure64-10.1.iso
-// |     +-- TinyCorePure64-10.1 (copy).iso
-// ...
-func TestGetCachedDirectory(t *testing.T) {
-	mp, err := getCachedDirectory()
-	if err != nil {
-		t.Fatalf("Fail to find the USB stick: %+v", err)
-	}
-	checklist := []string{
-		filepath.Join(mp.Path, "/Image/tinycore1/TinyCorePure64.iso"),
-		filepath.Join(mp.Path, "/Image/tinycore1/TinyCorePure64 (copy).iso"),
-		filepath.Join(mp.Path, "/Image/tinycore2/TinyCorePure64-10.1.iso"),
-		filepath.Join(mp.Path, "/Image/tinycore2/TinyCorePure64-10.1 (copy).iso"),
-	}
-	for _, path := range checklist {
-		if _, err = os.Stat(path); err != nil {
-			t.Fatalf("Get wrong cache directory, do not find file %s", path)
-		}
+func TestGoBackDir(t *testing.T) {
```

Comment:
Sorry I think I phrased my last comment incorrectly, I didn't mean take out the UI testing - I mean can you just test a simpler case where you call exec for a DirOption at the root level, go two levels down, choose the back option, and then confirm that the new DirOption that you're presenting is the correct one? or something like that

Reply:
Yes, I understand what you mean. but when I think it further, I find why I implementing the whole process of back option at test? Actually the process is, if a backOption is called, I return a DirOption with it's father's path right. and most of the options in this process have been tested in the previous test, like correctly call a Option in the menu is be tested by menu_test. and create a DirOption with certern path is be tested in the TestDirOption. So the rest logic that haven't be covered in the past test is only the goBack function. So I just test this function here is ok. Is that make sense?

Reply:
And I think that makes the test more easily to read and understand.

Reply:
I think it'd just be nice to have the UI test for this case, up to you :)

Reply:
emmm ok, I'll add the ui test too.